torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1146,6 +1146,14 @@ def _register_param_handles_from_handles(
                 prefixed_param_names.extend(old_handle.flat_param._prefixed_param_names)
                 flat_params.append(old_handle.flat_param)
             self._rebuild_full_params(flat_params)
+            # For each parameter in `flat_params`, its sharded tensors will
+            # not be used anymore. We explicitly free their storage for better
+            # memory efficiency.
+            for flat_param in flat_params:
+                if hasattr(flat_param, "_local_shard"):
```

Comment:
After we merge, I will refactor this into a `_free_param_attributes()` method (to complement `_init_param_attributes()`).

Reply:
Just kidding. It might not make sense to write a `_free_param_sharded_attributes()` since we are only freeing the sharded tensors. I will leave it as is right now and can revisit when we open the actual PR.